server/script/parser/grammar.lua

```diff
@@ -399,7 +399,9 @@ Generics1   <-  Sp ({} '<' Sp (Name / Sp COMMA {})+ Sp '>' {})
 Generics2   <-  Sp ({} '<' Sp (Type / Sp COMMA {})+ Sp '>' {})
             ->  Generics
 
-TypeList    <-  ({} PL (Type / Sp COMMA {})* NeedPR {} Optional)
+TypeIdTag   <-  ({} Name Sp COLON {})
+            ->  TypeIdTag
+TypeList    <-  ({} PL ((TypeIdTag? Sp Type) / Sp COMMA {})* NeedPR {} Optional)
             ->  TypeList
```

Comment:
```suggestion
FuncTypeList
            <-  ({} PL ((TypeIdTag? Sp Type) / Sp COMMA {})* NeedPR {} Optional)
            ->  TypeList
TypeList    <-  ({} PL (Type / Sp COMMA {})* NeedPR {} Optional)
            ->  TypeList
```

Reply:
TypeList is also used for multiple returns, add a new grammar called FuncTypeList and use it in FuncType.